File: 513.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def findBottomLeftValue(self, root: Optional[TreeNode]) -> int:
        self.res = root.val
        self.height = 0

        def rln(node, height):
            if node is None:
                return
            if height >= self.height:
                self.res = node.val
                self.height = height
            rln(node.right, height + 1)
            rln(node.left, height + 1)

        rln(root, 0)
        return self.res

File: sol/513/513.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
struct Solution {
    void nrl(TreeNode *node, int h, int &res, int &height){
        if (node == nullptr){
            return;
        }
        if (h >= height){
            height = h;
            res = node->val;
        }
        ++h;
        this->nrl(node->right, h, res, height);
        this->nrl(node->left, h, res, height);
    }
    int findBottomLeftValue(TreeNode* root) {
        int res = root->val, height = 0;
        this->nrl(root, 0, res, height);
        return res;
    }
};
